File: src/OpenCvSharp.Test/Samples/SeqTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;
using OpenCvSharp;

namespace OpenCvSharp.Test
{
    /// <summary>
    /// CvSeqのテスト
    /// </summary>
    class SeqTest
    {        
        public unsafe SeqTest()
        {
            using (CvMemStorage storage = new CvMemStorage(0))
            {
                Random rand = new Random();
                CvSeq<int> seq = new CvSeq<int>(SeqType.EltypeS32C1, storage);
                // push
                for (int i = 0; i < 10; i++)
                {
                    int push = seq.Push(rand.Next(100));//seq.Push(i);
                    Console.WriteLine("{0} is pushed", push);
                }
                Console.WriteLine("----------");

                // enumerate
                Console.WriteLine("contents of seq");
                foreach (int item in seq)
                {
                    Console.Write("{0} ", item);
                }
                Console.WriteLine();

                // sort
                CvCmpFunc<int> func = delegate(int a, int b)
                {
                    return a.CompareTo(b);
                };
                seq.Sort(func);

                // convert to array
                int[] array = seq.ToArray();
                Console.WriteLine("contents of sorted seq");
                foreach (int item in array)
                {
                    Console.Write("{0} ", item);
                }
                Console.WriteLine();
                Console.WriteLine("----------");

                // pop
                for (int i = 0; i < 10; i++)
                {
                    int pop = seq.Pop();
                    Console.WriteLine("{0} is popped", pop);
                }
                Console.ReadKey();
            }
        }
    }
}

File: sample/OpenCvSharpSamplesVB/Samples/SeqTest.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Runtime.InteropServices
Imports System.Text
Imports OpenCvSharp

' Namespace OpenCvSharpSamplesVB
    ''' <summary>
    ''' CvSeqのテスト
    ''' </summary>
    Friend Module SeqTest
        'INSTANT VB TODO TASK: There is no equivalent to the 'unsafe' modifier in VB:
        'ORIGINAL LINE: public unsafe SeqTest()
        Public Sub Start()
            Using storage As New CvMemStorage(0)
                Dim rand As New Random()
                Dim seq As New CvSeq(Of Integer)(SeqType.EltypeS32C1, storage)
                ' push
                For i As Integer = 0 To 9
                    Dim push As Integer = seq.Push(rand.Next(100)) 'seq.Push(i);
                    Form1.TextBox1.AppendText(Environment.NewLine & String.Format("{0} is pushed", push))
                Next i
                Form1.TextBox1.AppendText(Environment.NewLine & String.Format("----------"))

                ' enumerate
                Form1.TextBox1.AppendText(Environment.NewLine & String.Format("contents of seq"))
                For Each item As Integer In seq
                    Form1.TextBox1.AppendText(Environment.NewLine & String.Format("{0} ", item))
                Next item
                Form1.TextBox1.AppendText(Environment.NewLine & " ")

                ' sort
                Dim func As CvCmpFunc(Of Integer) = Function(a As Integer, b As Integer) a.CompareTo(b)
                seq.Sort(func)

                ' convert to array
                Dim array() As Integer = seq.ToArray()
                Form1.TextBox1.AppendText(Environment.NewLine & String.Format("contents of sorted seq"))
                For Each item As Integer In array
                    Form1.TextBox1.AppendText(Environment.NewLine & String.Format("{0} ", item))
                Next item
                Form1.TextBox1.AppendText(Environment.NewLine & " ")
                Form1.TextBox1.AppendText(Environment.NewLine & String.Format("----------"))

                ' pop
                For i As Integer = 0 To 9
                    Dim pop As Integer = seq.Pop()
                    Form1.TextBox1.AppendText(Environment.NewLine & String.Format("{0} is popped", pop))
                Next i
                'Console.ReadKey()
            End Using
        End Sub
    End Module
' End Namespace
